### PR TITLE
WIP: Cldc 600 reduce user burden 

### PR DESCRIPTION
**Pull request by dushan-madetech**

### CODE REVIEW

**Review by smeny — APPROVED** [review]
Everything looks fine to me but I don't really if there is a better way to add new fields given the existing structure. So you might want a second opinion from @baarkerlounger or @kosiakkatrina

File: spec/features/form/page_routing_spec.rb [hunk 1]
```diff
@@ -72,4 +72,28 @@
       expect(page).to have_current_path("/logs/#{id}/property-wheelchair-accessible")
     end
   end
+
+  context "routing based on start date period", js: true do
```
Comment (baarkerlounger) [inline spec/features/form/page_routing_spec.rb:76]: I think this functionality is well tested here already https://github.com/communitiesuk/mhclg-data-collection-beta/blob/main/spec/requests/form_controller_spec.rb#L55 so these feature ones might not be necessary?

File: spec/requests/case_log_controller_spec.rb [hunk 2]
```diff
@@ -240,6 +240,31 @@
...
+          FactoryBot.create(
+            :case_log,
+            :about_completed,
+            startdate: Time.zone.local(2022, 10, 10),
+            owning_organisation: organisation,
+            managing_organisation: organisation,
+          )
+        end
+
+        it "has a joint tenancy page that can be visited if the case log is in the 2022-2023 period" do
```
Comment (baarkerlounger) [inline spec/requests/case_log_controller_spec.rb:256]: Same with this spec. Not sure it's adding much over the one that's there already? Also I think it would probably belong in `form_controller_spec` (form navigation) rather than `case_log_controller_spec` (case log data API)

**Review by baarkerlounger — APPROVED** [review]
Didn't realise we were starting work on next years form already. Couple of comments on tests but nothing blocking.

File: app/helpers/tasklist_helper.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ def first_page_or_check_answers(subsection, case_log)
     path = if subsection.is_started?(case_log)
              "case_log_#{subsection.id}_check_answers_path"
            else
-             "case_log_#{subsection.pages.first.id}_path"
+             "case_log_#{subsection.applicable_questions(case_log).first.page.id}_path"
```
Comment (baarkerlounger) [inline app/helpers/tasklist_helper.rb:32]: Should there be a test for the case this change is covering? Or have I missed one?

**Comment by baarkerlounger** [comment]
I wonder if creating next year's form now (before completing this year's form) is going to make it harder to keep up to date/correct since every change PR now needs to be made against both form definition files.

It might be easier to complete this years and then do next years changes as a single PR?

File: spec/features/form/page_routing_spec.rb (hunk 1 above)
Comment (dushan-madetech) [reply spec/features/form/page_routing_spec.rb:76]: This was originally written prior to those tests I'm happy to remove since I think this is also covered by some request specs

File: spec/requests/case_log_controller_spec.rb (hunk 2 above)
Comment (dushan-madetech) [reply spec/requests/case_log_controller_spec.rb:256]: I think there's a slight difference where one is checking the difference in headers for the same question whereas this is checking the difference in presence of entire pages in a section. Happy to remove/move this though

Comment (baarkerlounger) [reply spec/requests/case_log_controller_spec.rb:256]: In both cases the behaviour actually being checked is that the correct form for that year is being routed to though right?

File: app/helpers/tasklist_helper.rb (hunk 3 above)
Comment (dushan-madetech) [reply app/helpers/tasklist_helper.rb:32]: I think this was originally covered by the tests added in this PR - I did recently have to rebase it and update the tests due to all the changes that happened while I was away. It's not breaking any existing tests at the moment so I'd be willing to say that's a promising sign

**Comment by dushan-madetech** [comment]
> I wonder if creating next year's form now (before completing this year's form) is going to make it harder to keep up to date/correct since every change PR now needs to be made against both form definition files.
> 
> It might be easier to complete this years and then do next years changes as a single PR?

@baarkerlounger I think I'm inclined to agree with you - I'll have a look at splitting the ticket into separate parts so the that the 2021-2022 part could be shipped now

File: spec/requests/case_log_controller_spec.rb (hunk 2 above)
Comment (dushan-madetech) [reply spec/requests/case_log_controller_spec.rb:256]: Yep, that's a fair point. Happy to remove these in that case

File: app/helpers/tasklist_helper.rb (hunk 3 above)
Comment (baarkerlounger) [reply app/helpers/tasklist_helper.rb:32]: Might be worth adding a unit test for the specific edge case here https://github.com/communitiesuk/mhclg-data-collection-beta/blob/main/spec/helpers/tasklist_helper_spec.rb#L44 but not a blocker.
